src/messages/MClientReply.h

```diff
@@ -343,11 +342,12 @@ class MClientReply final : public MMDSOp {
     // errorcode32_t is converting, internally, the error code from host to ceph, when encoding, and vice versa,
     // when decoding, resulting having LINUX codes on the wire, and HOST code on the receiver.
     // assumes this code is executing after decode_payload() function has been called
-    return error_code;
+    return head.result;
   }
 
-  void set_result(__s32 r) { //
-    error_code = r;
+  // errorcode32_t is used in decode/encode methods
+  void set_result(__s32 r) {
+    head.result = r;
```

Comment:
the above codes set the head.result to unsigned int but this helper casts to signed int.

Reply:
Well there is type hell here. `head.result` is defined as `__le32` which is eventually defined as "little endian unsigned 32bit". I assume this field is very fragile since it is dencoded between user and kernel, so I will not change it's type.
The rest of the code uses signed int for error handling (using different aliases, e.g. `__s32`, `int32_t`or just `int`, so somewhere the conversion between signed and unsigned should occur.
I am not sure that fragility of chaging the type of `head.result` worth the consistency of the code in this case.